Update Community Group Charter for 2024->

anssiko opened the pull request:
- Refresh Goals
- Add Task-specific APIs and Prompt API to Deliverables
- Note WebNN has graduated
- Stylistic tweaks

Note: This is a draft for discussion and review.

[Preview](https://raw.githack.com/webmachinelearning/charter/refs/heads/charter-2024/index.html)

anssiko committed:
Update Community Group Charter for 2024->

- Refresh Goals
- Add Task-specific APIs and Prompt API to Deliverables
- Note WebNN has graduated
- Stylistic tweaks

domenic commented on index.html:
```diff
@@ -113,6 +140,35 @@ <h3 id="neural-network-inference">
+      <li>Translation between languages;</li>
+      <li>Proofreading.</li>
+    </ul>
     <p>
```
Nit: these paragraphs are now under "prompt API". Maybe they should move up to the top?

domenic commented on index.html:
```diff
@@ -158,7 +214,25 @@ <h3 id="specifications">
+        A dedicated low-level API for neural network inference hardware acceleration.
+      </dd>
+      <dt>
+        Translator API
```
Maybe include "Translator and Language Detector" APIs in this bullet?

domenic reviewed (commented):
This looks great! Only minor comments from me.

anssiko committed:
Move generic Scope of Work statements to the top

anssiko committed:
Add Language Detector to Deliverables

anssiko commented:
@domenic thanks for the suggestions, PR updated accordingly.

anssiko committed:
Fix typo

HelloFillip commented on index.html:
```diff
@@ -113,16 +150,35 @@ <h3 id="neural-network-inference">
-      Each specification should detail all known security and privacy
-      implications for implementers, Web authors, and end users.
+      A general-purpose API to prompt a language model directly using common
+      promping techniques to accomplish a variety of tasks, including:
```
A note on this: is privacy and security being reduced / removed as a priority. As someone focused primarily on the privacy implications I would hope to ensure there’s still an emphasis on safety-by-design throughout the process.

anssiko committed:
Remove [DRAFT] tag, update Start Date

Charter review ended 2024-12-02 with explicit support,
no objections and only minor editorial changes to the
charter based on the review.

anssiko commented:
Thank you for your review and support everyone! 🚀 

I will merge this PR now with a note this new charter will be operational on 9 December 2024 per the charter amendment process.

manc95 reviewed (commented):
1
